addons/im_livechat/models/discuss_channel.py

```diff
@@ -545,6 +547,20 @@ def _email_livechat_transcript(self, email):
         })
         mail.send()
 
+    def _get_attachment_data(self, attachment):
+        file_extension = get_extension(attachment.display_name)
+        attachment_data = {
+            "id": attachment.id,
+            "extension": file_extension.lstrip("."),
+            "mimetype": attachment.mimetype,
+            "filename": attachment.display_name,
+            "url": attachment.url,
+        }
+        return self.env["ir.qweb"]._render(
+            "im_livechat.attachment_template",
+            {"props": json.dumps({"fileData": attachment_data})},
+        )
```

Comment:
Do we really need a template for a single div? Can't we build it here?
Do we want users to be able to customize it?

Reply:
IMO keeping it inside a template is better even if it's just a DIV. As it is XML code.